crowbar_framework/app/models/keystone_service.rb

```diff
@@ -94,6 +94,17 @@ def validate_proposal_after_save proposal
       validation_error("Keystone API version 3 needs to be enabled in order to use domain specific drivers.")
     end
 
+    keystone_timeout = proposal["attributes"]["keystone"]["token_expiration"]
+    horizon_proposal = ProposalObject.find_proposal("nova_dashboard", "default")
+    unless horizon_proposal.nil?
+      horizon_timeout = horizon_proposal["attributes"]["nova_dashboard"]["session_timeout"]
+
+      # keystone_timeout is in seconds and horizon_timeout is in minutes
+      if horizon_timeout * 60 > keystone_timeout
+        validation_error("Setting the Horizon timeout (#{horizon_timeout} minutes) longer than the Keystone token expiration timeout (#{keystone_timeout/60} minutes) is not supported.")
```

Comment:
Can you use the same code as in the crowbar-openstack PR? We don't have hound running here, but that line is too long :-)